### ISSUE TITLE
The public files under project root were not copyed to dist

### ISSUE THREAD

**Issue by bangbangde** [vuejs/vitepress#2053]
### Describe the bug

I setted the `config.srcDir: 'docs'` and runs the cmd `npm run docs:build` (`vitepress build`).

but the public files under project root were not copyed to dist as described at [guide](https://vitepress.vuejs.org/guide/asset-handling#public-files).

I found that the build cmd looks for `publicDir` under `config.srcDir`.

### Reproduction

[stackblitz](https://stackblitz.com/edit/vite-kjqd4c)

runs `npm run docs:build`

### Expected behavior

Assets placed in public be copied to the root of the dist directory as-is.

### System Info

```shell
System:
    OS: macOS 13.2.1
    CPU: (4) x64 Intel(R) Core(TM) i5-8210Y CPU @ 1.60GHz
    Memory: 21.66 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.4.0 - ~/.nvm/versions/node/v18.4.0/bin/node
    npm: 8.12.1 - ~/.nvm/versions/node/v18.4.0/bin/npm
  Browsers:
    Chrome: 110.0.5481.177
    Safari: 16.3
  npmPackages:
    vitepress: ^1.0.0-alpha.49 => 1.0.0-alpha.49
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.vuejs.org).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by bangbangde** [comment]
I think the behavior described in the document is more intuitive.

**Comment by brc-dd** [comment]
I think it's expected behavior. `public`, `vite.config.ts`, etc. all should be inside the `srcDir` and that `srcDir` is resolved relative to `root`. So, suppose you've a Vite project and want to keep `.vitepress` at `.` and set `srcDir: 'docs'`, now if it picks `vite.config.ts` it's gonna be a problem as that file was never meant for `docs`.

**Comment by yyx990803** [comment]
This is a documentation error.

The `public` dir is indeed relative to `srcDir`, because `srcDir` is where the dev server / routing considers to be its root.

**Comment by yyx990803** [comment]
fixed in 31b548c
